source/DSMainWindow.cpp

```diff
@@ -17,21 +19,23 @@ void DSMainWindow::createActions() {
         for(int i = 0; i < list->size(); i++) tool_bar->addAction(list->at(i));
         tool_bar->insertSeparator(list->at(n));
         */
+    createActionLoadVoice();
+    createActionShowVoicePath();
 }
 
 void DSMainWindow::createMenus() {
-    /* EXAMPLE CODE
         QMenu* fileMenu = new QMenu("File", this);
-        for(int i = 0; i < list->size(); i++) {
-            fileMenu->addAction(list->at(i));
+        for(int i = 0; i < actions.size(); i++) {
+            fileMenu->addAction(actions.at(i));
             // if(i == 2 || i == 6) fileMenu->addSeparator();
         }
         fileMenu->addSeparator();
+
         QAction* exitAct = new QAction(QIcon(":/Risorse/Icons/exit.png"), "Esci", this);
         fileMenu->addAction(exitAct);
-        connect(exitAct, &QAction::triggered, this, &MainWindow::shutDownApp);
+        connect(exitAct, &QAction::triggered, this, &DSMainWindow::close);
```

Comment:
Tutte le connect le raggrupperei in una funzione sola per ogni classe, di nome doConnections; quindi sposterei questa istruzione dentro DSMainWindow::doConnections

Reply:
il problema di fare come hai detto tu è che le QAction sono salvate in un vettore, quindi se si fa in un'altra funzione bisognerebbe non si sa che posizione del vettore va collegata a che evento